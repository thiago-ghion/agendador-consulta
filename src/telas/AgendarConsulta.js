import React, { useState, useRef, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import moment from "moment";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Form from "react-bootstrap/Form";
import { Button } from "react-bootstrap";
import { isMobile } from "react-device-detect";

import BootstrapTable from "react-bootstrap-table-next";
import ToolkitProvider from "react-bootstrap-table2-toolkit/dist/react-bootstrap-table2-toolkit.min";

import { AsyncTypeahead, Typeahead } from "react-bootstrap-typeahead";
import "react-bootstrap-typeahead/css/Typeahead.css";
import DatePicker from "../componentes/DatePicker";

import { setSucesso } from "../features/mensagemSlice";
import {
  limparListaPaciente,
  listarParcialPacienteAction,
} from "../features/pacienteSlice";
import {
  listarVigenteAction,
  limparListaDataDisponivel,
  listarDataDisponivelAction,
  listarHorarioDisponivelAction,
  limparListaHorarioDisponivel,
} from "../features/ProfissionalSlice";
import { agendarAction } from "../features/consultaSlice";

function AgendarConsulta() {
  const dispatch = useDispatch();

  const { listaPaciente } = useSelector((state) => state.paciente);
  const {
    listaProfissionalVigente,
    listaDataDisponivel,
    listaHorarioDisponivel,
  } = useSelector((state) => state.profissional);

  const [isLoading, setIsLoading] = useState(false);
  const [isGradeDataCarregada, setIsGradeDataCarregada] = useState(false);
  const [isGradeHorarioCarregada, setIsGradeHorarioCarregada] = useState(false);
  const [paciente, setPaciente] = useState(null);
  const [profissional, setProfissional] = useState(null);
  const [comboPaciente, setComboPaciente] = useState([]);
  const [comboProfissional, setComboProfissional] = useState([]);

  const [data, setData] = useState(null);

  const isTelaCarregada = useRef(false);

  useEffect(() => {
    if (!isTelaCarregada.current) {
      dispatch(listarVigenteAction());
      dispatch(limparListaPaciente());
      dispatch(limparListaDataDisponivel());
      dispatch(limparListaHorarioDisponivel());
    }
    isTelaCarregada.current = true;
  });

  const filterBy = () => true;

  const procuraPaciente = async (nomeParcial) => {
    setIsLoading(true);
    await dispatch(listarParcialPacienteAction({ nomeParcial }));
    setIsLoading(false);
  };

  const carregarGradeData = async (profissional) => {
    setComboProfissional(profissional);
    dispatch(setSucesso(null));
    setProfissional(profissional[0]);
    const resposta = await dispatch(
      listarDataDisponivelAction({
        idProfissional: profissional[0].idProfissional,
        dataInicio: moment().format("DD.MM.YYYY"),
        dataFim: moment().startOf("day").add(60, "days").format("DD.MM.YYYY"),
      })
    );
    if (resposta.error === undefined) {
      setIsGradeDataCarregada(true);
    }
  };

  const isDataValida = (currentDate, selectedDate) => {
    let isValido = false;
    listaDataDisponivel.forEach((item) => {
      if (item.data === currentDate.format("DD.MM.YYYY")) {
        isValido = true;
      }
    });
    return isValido;
  };

  const carregarPaciente = (evento) => {
    dispatch(setSucesso(null));
    setPaciente(evento[0]);
    setComboPaciente(evento);
  };

  const carregarListaHorario = async (evento) => {
    setIsGradeHorarioCarregada(false);
    const dataPesquisa = evento.format("DD.MM.YYYY");
    const resposta = await dispatch(
      listarHorarioDisponivelAction({
        idProfissional: profissional.idProfissional,
        dataPesquisa,
      })
    );

    if (resposta.error === undefined) {
      setIsGradeHorarioCarregada(true);
    }
  };

  const agendar = (row) => async () => {
    const resposta = await dispatch(
      agendarAction({
        idPaciente: paciente.idPaciente,
        idProfissional: profissional.idProfissional,
        dataConsulta: data.format("DD.MM.YYYY"),
        idHorario: row.idHorario,
      })
    );
    if (resposta.error === undefined) {
      window.scrollTo(0, 0);
      dispatch(setSucesso("Consulta agenda com sucesso"));
      dispatch(limparListaPaciente());
      dispatch(limparListaDataDisponivel());
      dispatch(limparListaHorarioDisponivel());
      setIsGradeDataCarregada(false);
      setIsGradeHorarioCarregada(false);
      setComboPaciente([]);
      setComboProfissional([]);
    }
  };

  const horarioFormatter = (cell, row, rowIndex, formatExtraData) => {
    return (
      <center>
        <Button variant="primary" onClick={agendar(row)}>
          Agendar
        </Button>
      </center>
    );
  };

  const colunasHorario = [
    {
      dataField: "horario",
      text: "Horário",
      style: { cursor: "pointer" },
      sort: true,
      headerStyle: () => {
        return { width: "110px" };
      },
    },
    {
      dataField: "idHorario",
      text: "Ações",
      sort: true,
      style: { cursor: "pointer" },
      formatter: horarioFormatter,
    },
  ];

  return (
    <div>
      <br></br>
      <Row>
        <Col>
          <Form.Label>Paciente</Form.Label>
          <AsyncTypeahead
            filterBy={filterBy}
            id="paciente"
            isLoading={isLoading}
            onSearch={procuraPaciente}
            labelKey="nomePaciente"
            options={listaPaciente}
            minLength={3}
            placeholder="Escolha o paciente..."
            onChange={carregarPaciente}
            selected={comboPaciente}
          />
        </Col>
      </Row>
      <br></br>
      <Row>
        <Col>
          <Form.Label>Profissional</Form.Label>
          <Typeahead
            filterBy={filterBy}
            id="profissional"
            dropup={true}
            labelKey="nomeProfissional"
            options={listaProfissionalVigente}
            placeholder="Escolha o profissional..."
            onChange={carregarGradeData}
            selected={comboProfissional}
          />
        </Col>
      </Row>
      <br></br>
      {isGradeDataCarregada ? (
        <Row data-testid="blocoData">
          <Col>
            <DatePicker
              input={false}
              isValidDate={isDataValida}
              value={data}
              onChange={(evento) => {
                dispatch(setSucesso(null));
                setData(evento);
                carregarListaHorario(evento);
              }}
            ></DatePicker>
          </Col>
        </Row>
      ) : (
        <></>
      )}
      {isGradeHorarioCarregada ? (
        <div>
          <br></br>
          <Row>
            {isMobile ? <></> : <Col></Col>}
            <Col>
              <ToolkitProvider
                keyField="horario"
                data={listaHorarioDisponivel}
                columns={colunasHorario}
              >
                {(props) => (
                  <div>
                    <BootstrapTable
                      classes="react-bootstrap-table"
                      keyField="horario"
                      striped={true}
                      hover={true}
                      {...props.baseProps}
                    />
                  </div>
                )}
              </ToolkitProvider>
            </Col>
            {isMobile ? <></> : <Col></Col>}
          </Row>
        </div>
      ) : (
        <></>
      )}
    </div>
  );
}

export default AgendarConsulta;
