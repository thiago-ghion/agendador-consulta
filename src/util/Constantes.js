const URL_API = 'http://localhost:5000';

export { URL_API };