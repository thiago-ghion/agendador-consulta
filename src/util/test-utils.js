import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import loginReducer from "../features/loginSlice";
import profissionalReducer from "../features/profissionalSlice";
import consultaReducer from "../features/consultaSlice";
import pacienteReducer from "../features/pacienteSlice";
import horarioReducer from "../features/horarioSlice";
import usuarioReducer from "../features/usuarioSlice";
import loadingReducer from "../features/loadingSlice";
import mensagemgReducer from "../features/mensagemSlice";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        login: loginReducer,
        consulta: consultaReducer,
        profissional: profissionalReducer,
        paciente: pacienteReducer,
        horario: horarioReducer,
        usuario: usuarioReducer,
        loading: loadingReducer,
        mensagem: mensagemgReducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
