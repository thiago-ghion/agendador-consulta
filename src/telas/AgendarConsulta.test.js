import React from "react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { renderWithProviders } from "../util/test-utils";

import AgendarConsulta from "./AgendarConsulta";
import { fireEvent, screen, waitFor, within } from "@testing-library/react";
import { default as userEvent } from "@testing-library/user-event";
import moment from "moment";

let mockListaProfissionalVigente = [];

let estadoListaDataDisponivel = 200;
let mockListaDataDisponivel = [];

let estadoListaHoraDisponivel = 200;
let mockListaHoraDisponivel = [];

let estadoListaParcial = 200;
let mockListaParcial = [];

let estadoAgendar = 200;
let mockAgendar = {};

export const handlers = [
  rest.get(
    "http://localhost:5000/v1/profissional/listarVigente",
    (req, res, ctx) => {
      return res(ctx.json(mockListaProfissionalVigente));
    }
  ),
  rest.get(
    "http://localhost:5000/v1/profissional/listarDataDisponivel",
    (req, res, ctx) => {
      return res(
        ctx.status(estadoListaDataDisponivel),
        ctx.json(mockListaDataDisponivel)
      );
    }
  ),
  rest.get(
    "http://localhost:5000/v1/profissional/listarHorarioDisponivel",
    (req, res, ctx) => {
      return res(
        ctx.status(estadoListaHoraDisponivel),
        ctx.json(mockListaHoraDisponivel)
      );
    }
  ),
  rest.get(
    "http://localhost:5000/v1/paciente/listarParcial",
    (req, res, ctx) => {
      return res(ctx.status(estadoListaParcial), ctx.json(mockListaParcial));
    }
  ),
  rest.post("http://localhost:5000/v1/consulta/agendar", (req, res, ctx) => {
    return res(ctx.status(estadoAgendar), ctx.json(mockAgendar));
  }),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

beforeEach(() => {
  mockListaProfissionalVigente = [];
  estadoListaDataDisponivel = 200;
  mockListaDataDisponivel = [];
  estadoListaHoraDisponivel = 200;
  mockListaHoraDisponivel = [];
  estadoListaParcial = 200;
  mockListaParcial = [];
  estadoAgendar = 200;
  mockAgendar = {};
});

test("Renderização componente", async () => {
  expect(renderWithProviders(<AgendarConsulta />)).toMatchSnapshot();
});

test("Selecionar paciente", async () => {
  mockListaParcial = [{ idPaciente: 1, nomePaciente: "Jean Baumer" }];
  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const paciente = screen.getByPlaceholderText("Escolha o paciente...");

  await user.type(paciente, "Jean");

  await waitFor(() => new Promise((r) => setTimeout(r, 950)));

  const lista = await waitFor(() => screen.findByRole("option", /Jean/i));
  await user.click(lista);
});

test("Selecionar profissional - sem falha", async () => {
  mockListaProfissionalVigente = [
    { idProfissional: 1, nomeProfissional: "Maria da Silva" },
  ];
  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const profissional = screen.getByPlaceholderText("Escolha o profissional...");

  await user.type(profissional, "Maria", { delay: 20 });
  await user.click(screen.getByRole("option", /Maria/i));
});

test("Selecionar profissional - com falha", async () => {
  estadoListaDataDisponivel = 400;
  mockListaProfissionalVigente = [
    { idProfissional: 1, nomeProfissional: "Maria da Silva" },
  ];
  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const profissional = screen.getByPlaceholderText("Escolha o profissional...");

  await user.type(profissional, "Maria", { delay: 20 });
  await user.click(screen.getByRole("option", /Maria/i));
});

test("Selecionar data na grade de horário", async () => {
  mockListaProfissionalVigente = [
    { idProfissional: 1, nomeProfissional: "Maria da Silva" },
  ];
  mockListaDataDisponivel = [
    { data: moment().format("DD.MM.YYYY") },
    { data: moment().add(1, "day").format("DD.MM.YYYY") },
  ];
  mockListaHoraDisponivel = [{ horario: "08:00" }, { horario: "08:30" }];
  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const profissional = screen.getByPlaceholderText("Escolha o profissional...");

  await user.type(profissional, "Maria", { delay: 20 });
  await user.click(screen.getByRole("option", /Maria/i));
  const blocoData = await waitFor(() => screen.findByTestId("blocoData"));

  const celulas = await waitFor(() =>
    within(blocoData).findAllByText(moment().get("date"))
  );

  let celula;

  celulas.forEach((item) => {
    if (
      item.getAttribute("data-value") == moment().get("date") &&
      item.getAttribute("data-month") == moment().get("month") &&
      item.getAttribute("data-year") == moment().get("year")
    )
      celula = item;
  });

  expect(celula).not.toBeNull();
  await user.click(celula);

  await waitFor(() => screen.findByText("08:00"));
});

test("Carregamento da grade de horário com falha", async () => {
  mockListaProfissionalVigente = [
    { idProfissional: 1, nomeProfissional: "Maria da Silva" },
  ];
  mockListaDataDisponivel = [
    { data: moment().format("DD.MM.YYYY") },
    { data: moment().add(1, "day").format("DD.MM.YYYY") },
  ];
  estadoListaHoraDisponivel = 400;
  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const profissional = screen.getByPlaceholderText("Escolha o profissional...");

  await user.type(profissional, "Maria", { delay: 20 });
  await user.click(screen.getByRole("option", /Maria/i));
  const blocoData = await waitFor(() => screen.findByTestId("blocoData"));

  const celulas = await waitFor(() =>
    within(blocoData).findAllByText(moment().get("date"))
  );

  let celula;

  celulas.forEach((item) => {
    if (
      item.getAttribute("data-value") == moment().get("date") &&
      item.getAttribute("data-month") == moment().get("month") &&
      item.getAttribute("data-year") == moment().get("year")
    )
      celula = item;
  });

  expect(celula).not.toBeNull();
  await user.click(celula);

  await waitFor(() => {
    expect(screen.queryByText("08:00")).not.toBeInTheDocument();
  });
});

test("Agendar consulta - sucesso", async () => {
  mockListaParcial = [{ idPaciente: 1, nomePaciente: "Jean Baumer" }];
  mockListaProfissionalVigente = [
    { idProfissional: 1, nomeProfissional: "Maria da Silva" },
  ];
  mockListaDataDisponivel = [
    { data: moment().format("DD.MM.YYYY") },
    { data: moment().add(1, "day").format("DD.MM.YYYY") },
  ];
  mockListaHoraDisponivel = [{ horario: "08:00" }];

  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const paciente = screen.getByPlaceholderText("Escolha o paciente...");

  await user.type(paciente, "Jean");

  await waitFor(() => new Promise((r) => setTimeout(r, 950)));

  const lista = await waitFor(() => screen.findByRole("option", /Jean/i));
  await user.click(lista);

  const profissional = screen.getByPlaceholderText("Escolha o profissional...");

  await user.type(profissional, "Maria", { delay: 20 });
  await user.click(screen.getByRole("option", /Maria/i));
  const blocoData = await waitFor(() => screen.findByTestId("blocoData"));

  const celulas = await waitFor(() =>
    within(blocoData).findAllByText(moment().get("date"))
  );

  let celula;

  celulas.forEach((item) => {
    if (
      item.getAttribute("data-value") == moment().get("date") &&
      item.getAttribute("data-month") == moment().get("month") &&
      item.getAttribute("data-year") == moment().get("year")
    )
      celula = item;
  });

  expect(celula).not.toBeNull();
  await user.click(celula);

  await waitFor(() => screen.findByText("08:00"));

  await user.click(screen.queryByRole("button"), { name: /Agendar/i });
});

test("Agendar consulta - falha", async () => {
  estadoAgendar = 400;
  mockListaParcial = [{ idPaciente: 1, nomePaciente: "Jean Baumer" }];
  mockListaProfissionalVigente = [
    { idProfissional: 1, nomeProfissional: "Maria da Silva" },
  ];
  mockListaDataDisponivel = [
    { data: moment().format("DD.MM.YYYY") },
    { data: moment().add(1, "day").format("DD.MM.YYYY") },
  ];
  mockListaHoraDisponivel = [{ horario: "08:00" }];

  let user = userEvent.setup();
  renderWithProviders(<AgendarConsulta />);

  const paciente = screen.getByPlaceholderText("Escolha o paciente...");

  await user.type(paciente, "Jean");

  await waitFor(() => new Promise((r) => setTimeout(r, 950)));

  const lista = await waitFor(() => screen.findByRole("option", /Jean/i));
  await user.click(lista);

  const profissional = screen.getByPlaceholderText("Escolha o profissional...");

  await user.type(profissional, "Maria", { delay: 20 });
  await user.click(screen.getByRole("option", /Maria/i));
  const blocoData = await waitFor(() => screen.findByTestId("blocoData"));

  const celulas = await waitFor(() =>
    within(blocoData).findAllByText(moment().get("date"))
  );

  let celula;

  celulas.forEach((item) => {
    if (
      item.getAttribute("data-value") == moment().get("date") &&
      item.getAttribute("data-month") == moment().get("month") &&
      item.getAttribute("data-year") == moment().get("year")
    )
      celula = item;
  });

  expect(celula).not.toBeNull();
  await user.click(celula);

  await waitFor(() => screen.findByText("08:00"));

  await user.click(screen.queryByRole("button"), { name: /Agendar/i });
});
