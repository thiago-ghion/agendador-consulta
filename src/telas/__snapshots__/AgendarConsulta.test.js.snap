// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização componente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <br />
        <div
          class="row"
        >
          <div
            class="col"
          >
            <label
              class="form-label"
            >
              Paciente
            </label>
            <div
              class="rbt"
              style="outline: none; position: relative;"
              tabindex="-1"
            >
              <div
                style="display: flex; flex: 1; height: 100%; position: relative;"
              >
                <input
                  aria-autocomplete="both"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  autocomplete="off"
                  class="rbt-input-main form-control rbt-input"
                  placeholder="Escolha o paciente..."
                  role="combobox"
                  type="text"
                  value=""
                />
                <input
                  aria-hidden="true"
                  class="rbt-input-hint"
                  readonly=""
                  style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 2px 2px 2px 2px; line-height: normal; padding: 1px 1px 1px 1px;"
                  tabindex="-1"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <br />
        <div
          class="row"
        >
          <div
            class="col"
          >
            <label
              class="form-label"
            >
              Profissional
            </label>
            <div
              class="rbt"
              style="outline: none; position: relative;"
              tabindex="-1"
            >
              <div
                style="display: flex; flex: 1; height: 100%; position: relative;"
              >
                <input
                  aria-autocomplete="both"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  autocomplete="off"
                  class="rbt-input-main form-control rbt-input"
                  placeholder="Escolha o profissional..."
                  role="combobox"
                  type="text"
                  value=""
                />
                <input
                  aria-hidden="true"
                  class="rbt-input-hint"
                  readonly=""
                  style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 2px 2px 2px 2px; line-height: normal; padding: 1px 1px 1px 1px;"
                  tabindex="-1"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <br />
      <div
        class="row"
      >
        <div
          class="col"
        >
          <label
            class="form-label"
          >
            Paciente
          </label>
          <div
            class="rbt"
            style="outline: none; position: relative;"
            tabindex="-1"
          >
            <div
              style="display: flex; flex: 1; height: 100%; position: relative;"
            >
              <input
                aria-autocomplete="both"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="rbt-input-main form-control rbt-input"
                placeholder="Escolha o paciente..."
                role="combobox"
                type="text"
                value=""
              />
              <input
                aria-hidden="true"
                class="rbt-input-hint"
                readonly=""
                style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 2px 2px 2px 2px; line-height: normal; padding: 1px 1px 1px 1px;"
                tabindex="-1"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <br />
      <div
        class="row"
      >
        <div
          class="col"
        >
          <label
            class="form-label"
          >
            Profissional
          </label>
          <div
            class="rbt"
            style="outline: none; position: relative;"
            tabindex="-1"
          >
            <div
              style="display: flex; flex: 1; height: 100%; position: relative;"
            >
              <input
                aria-autocomplete="both"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="rbt-input-main form-control rbt-input"
                placeholder="Escolha o profissional..."
                role="combobox"
                type="text"
                value=""
              />
              <input
                aria-hidden="true"
                class="rbt-input-hint"
                readonly=""
                style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 2px 2px 2px 2px; line-height: normal; padding: 1px 1px 1px 1px;"
                tabindex="-1"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
